src/backend/curses/pan.rs

```diff
@@ -0,0 +1,218 @@
+extern crate pancurses;
+
+use backend;
+use event::{Event, Key};
+
+use self::super::find_closest;
+use theme::{Color, ColorStyle, Effect};
+
+pub struct Concrete {
+    window: pancurses::Window,
+}
+
+impl backend::Backend for Concrete {
+    fn init() -> Self {
+        ::std::env::set_var("ESCDELAY", "25");
+        let window = pancurses::initscr();
+        window.keypad(true);
+        pancurses::noecho();
+        pancurses::cbreak();
+        pancurses::start_color();
+        pancurses::curs_set(0);
+        window.bkgd(pancurses::COLOR_PAIR(ColorStyle::Background.id() as u64));
+
+        Concrete { window: window }
+    }
+
+    fn screen_size(&self) -> (usize, usize) {
+        let (x, y) = self.window.get_max_yx();
+        (x as usize, y as usize)
+    }
+
+    fn has_colors(&self) -> bool {
+        pancurses::has_colors()
+    }
+
+    fn finish(&mut self) {
+        pancurses::endwin();
+    }
+
+    fn init_color_style(&mut self, style: ColorStyle, foreground: &Color,
+                        background: &Color) {
+        pancurses::init_pair(style.id(),
+                             find_closest(foreground) as i16,
+                             find_closest(background) as i16);
+    }
+
+    fn with_color<F: FnOnce()>(&self, color: ColorStyle, f: F) {
+        // TODO: pancurses doesn't have an `attr_get` equivalent
+        // let mut current_style: pancurses::attr_t = 0;
+        // let mut current_color: i16 = 0;
+        // pancurses::attr_get(&mut current_style, &mut current_color);
```

Comment:
Though well think about that after this PR, as the mechanism may go into `Printer` instead.